test/BatchExchangeViewer.spec.ts

```diff
@@ -531,4 +532,96 @@ contract("BatchExchangeViewer [ @skip-on-coverage ]", (accounts) => {
+      );
+      await token_1.givenMethodReturnUint(decimalsMethod, 42);
+
+      const viewer = await BatchExchangeViewer.new(batchExchange.address);
+      const result = await viewer.getTokenInfo(1);
+      assert.equal(result[0], token_1.address);
+      assert.equal(result[1], "SCAM");
+      assert.equal(result[2].toNumber(), 42);
+    });
+    it("Returns unknown if token doesn't implement symbol", async () => {
+      const erc20detailed = await ERC20Detailed.at(token_1.address);
+      const symbolMethod = erc20detailed.contract.methods.symbol().encodeABI();
+      await token_1.givenMethodRevert(symbolMethod);
+
+      const decimalsMethod = erc20detailed.contract.methods
+        .decimals()
+        .encodeABI();
+      await token_1.givenMethodReturnUint(decimalsMethod, 42);
+
+      const viewer = await BatchExchangeViewer.new(batchExchange.address);
+      const result = await viewer.getTokenInfo(1);
+      assert.equal(result[0], token_1.address);
+      assert.equal(result[1], "Unknown");
+      assert.equal(result[2].toNumber(), 42);
+    });
+    it("Reverts if token doesn't implement decimals", async () => {
+      const erc20detailed = await ERC20Detailed.at(token_1.address);
+      const symbolMethod = erc20detailed.contract.methods.symbol().encodeABI();
+      await token_1.givenMethodReturn(
+        symbolMethod,
+        web3.eth.abi.encodeParameter("string", "SCAM"),
+      );
+
+      const decimalsMethod = erc20detailed.contract.methods
+        .decimals()
+        .encodeABI();
+      await token_1.givenMethodRevert(decimalsMethod);
+
+      const viewer = await BatchExchangeViewer.new(batchExchange.address);
+      await truffleAssert.reverts(viewer.getTokenInfo(1));
+    });
+    it("Can fetch MKR token ", async () => {
+      const erc20detailed = await ERC20Detailed.at(token_1.address);
+      const symbolMethod = erc20detailed.contract.methods.symbol().encodeABI();
+      // cf. https://etherscan.io/token/0x9f8f72aa9304c8b593d555f12ef6589cc3a579a2#readContract
+      await token_1.givenMethodReturn(
+        symbolMethod,
+        "0x4d4b520000000000000000000000000000000000000000000000000000000000",
+      );
+
+      const decimalsMethod = erc20detailed.contract.methods
+        .decimals()
+        .encodeABI();
+      await token_1.givenMethodReturnUint(decimalsMethod, 18);
+
+      const viewer = await BatchExchangeViewer.new(batchExchange.address);
+      const result = await viewer.getTokenInfo(1);
+      assert.equal(result[0], token_1.address);
+      assert.equal(result[1], "MKR");
+      assert.equal(result[2].toNumber(), 18);
+    });
+    it("Can fetch dynamic byte arrays ", async () => {
+      const erc20detailed = await ERC20Detailed.at(token_1.address);
+      const symbolMethod = erc20detailed.contract.methods.symbol().encodeABI();
+      await token_1.givenMethodReturn(
+        symbolMethod,
+        web3.eth.abi.encodeParameter("bytes", "0x464f4f"),
```

Comment:
Oops, sorry for the extra work. I didn't realize that the strings and dynamic byte arrays actually encode in the exact same way:
```
> web3.eth.abi.encodeParameter("bytes", "0x464f4f") === web3.eth.abi.encodeParameter("string", "FOO")
true
```